dvc/repo/index.py

```diff
@@ -91,6 +91,13 @@ def __contains__(self, stage: "Stage") -> bool:
     def __iter__(self) -> Iterator["Stage"]:
         yield from self.stages
 
+    def __getitem__(self, item: str) -> "Stage":
+        """Get a Stage by its addressing attribute"""
+        for stage in self.stages:
```

Comment:
```suggestion
        for stage in self:
```

Reply:
We could also think of making `self.stages` a `dict` in the future, but since we don't use it in DVC, it's fine for now.